tests/gtest/CMakeLists.txt

```diff
@@ -1,4 +1,10 @@
-set(GTEST_ROOT_DIR ${TR_THIRD_PARTY_SOURCE_DIR}/googletest/googletest)
+find_path(GTEST_ROOT_DIR
+    NAMES
+        src/gtest-all.cc
+        src/gtest_main.cc
+    HINTS
+        /usr/src/googletest/googletest
+        ${TR_THIRD_PARTY_SOURCE_DIR}/googletest/googletest)
```

Comment:
Hi @niol 
This change is breaking the CI pipelines here:
https://github.com/transmission/transmission/actions/runs/9407489483/job/25913360961?pr=6900
```
  CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
  Please set them or make sure they are set and tested correctly in the CMake files:
  /home/runner/work/transmission/transmission/tests/gtest/GTEST_ROOT_DIR
     used as include directory in directory /home/runner/work/transmission/transmission/tests/gtest
  
  CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:639 (_add_library):
Error:     Cannot find source file:
  
      GTEST_ROOT_DIR-NOTFOUND/src/gtest-all.cc
```

And here:
https://trtc.mikedld.com/buildConfiguration/Transmission_Sanity/411238?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandchucktcpluginBuild=true&expandPull+Request+Details=true&expandBuildProblemsSection=true&expandBuildChangesSection=true
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/home/user/buildAgent/work/1f3bff43dd55cfe9/tests/gtest/GTEST_ROOT_DIR
   used as include directory in directory /home/user/buildAgent/work/1f3bff43dd55cfe9/tests/gtest
CMake Error at tests/gtest/CMakeLists.txt:9 (add_library):
  Cannot find source file:
    GTEST_ROOT_DIR-NOTFOUND/src/gtest-all.cc
  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
CMake Error at tests/gtest/CMakeLists.txt:9 (add_library):
  No SOURCES given to target: gtestall
```

You may have to address it or revert that part before admins can review your pull request.

Reply:
I have no clue of why this fails only on android, I need to understand what is different in this environment.

Reply:
It also fails on Gentoo/s390x (the second link with the TeamCity pipeline from mikedld.com)